### ISSUE TITLE
Catching exceptions in async code does not work anymore

### ISSUE THREAD

**Issue by andreaferretti** [nim-lang/Nim#4196]
I was using [this technique](http://forum.nim-lang.org/t/2129) to handle errors in async code:

```nim
var fut = asyncProc()
await fut
if fut.failed:
  echo("The async proc raised an error")
```

This seems to be broken somewhere between [this](https://github.com/nim-lang/Nim/commit/f4f7edf00f19e5abcd4f3a4631261e109d6b8c4d) and [this](https://github.com/nim-lang/Nim/commit/268f25225abab72e197274c31ebfee7ab6dbe0d4) commit.

A cursory glance at that range makes me think that the culprit is [this](https://github.com/nim-lang/Nim/commit/aba60e54d58a0425ae707efef4e3dfcf1a54064d), since I do not see other commits related to async (but I may have missed some).

I was using this in [rosencrantz](https://github.com/andreaferretti/rosencrantz) to catch exceptions and return an error 500 instead of crashing the whole process, so at the moment any exception in rosecrantz will bring the whole server down

**Comment by dom96** [comment]
Duplicate of https://github.com/nim-lang/Nim/issues/4170

**Comment by dom96** [comment]
I don't think it's to do with that commit, you sure it works before https://github.com/nim-lang/Nim/commit/f4f7edf00f19e5abcd4f3a4631261e109d6b8c4d ?

**Comment by andreaferretti** [comment]
Well, yesterday at home I updated and it broke. So before opening the bug I checked at work that everything was still ok, took note of the commit I was at, updated and watched it break again. So it was somewhere between that commit and today.

I can testify it was working ok regularly in the last weeks, since I tested rosencrantz every few days.

If you want to test a particular version, just checkout [rosencrantz](https://github.com/andreaferretti/rosencrantz), then in one tab do `nimble server` and in the other one `nimble client`.

Right now the tests are passing, but if you decomment [these lines](https://github.com/andreaferretti/rosencrantz/blob/master/tests/client.nim#L128-L131) it breaks
